packages/driver/cypress/e2e/e2e/origin/cookie_misc.cy.ts

```diff
@@ -205,4 +207,56 @@ describe('misc cookie tests', { browser: '!webkit' }, () => {
       })
     })
   })
+
+  describe('Same-Site Cross-Origin cookie behavior', () => {
+    const cookie = 'foo1=bar2'
+    const site = 'foobar.com'
+    const hostnameA = `www.${site}:3500`
+    const hostnameB = `app.${site}:3500`
+
+    beforeEach(() => {
+      cy.intercept(`http://${hostnameA}/test-request`, (req) => {
+        req.reply({
+          statusCode: 200,
+        })
+      }).as('cookiedRequest')
+    })
+
+    it('attaches cookies to proxied requests in a same-site cross-origin context', () => {
+      // 1. visits www.foobar.com:3500: The AUT and top are the same-origin
+      cy.visit(`http://${hostnameA}`).then(() => {
+        // 2. set a cookie via Set-Cookie Response header. Since top and AUT are the same-origin, this works
+        // @ts-expect-error
+        cy.wrap(makeRequest(window.top, `http://${hostnameA}/set-cookie?cookie=${cookie}; Domain=${site}`, 'fetch'))
+      })
+
+      // 3. navigate the AUT to app.foobar.com:3500. Now AUT and top are same-site, but cross-origin.
+      // in an actual browser, the AUT and top would both be app.foobar.com:3500 and would be same-origin
+      cy.visit(`http://${hostnameB}`).then(() => {
+        // 4. set a cookie via Set-Cookie Response header. This FAILS in Cypress <= 13 here because the AUT is cross-origin to top,
+        // but WOULD PASS in an actual browser because AUT and top would both be same-origin.
+
+        // This also does NOT get simulated in Cypress <= 13 because the simulation conditions in the Cypress middleware check
```

Comment:
wonder if we should change these comments to be more along the line of `injectDocumentDomain=true` but it feels weird tying document domain injection specifically to cookie behavior so I'm OK with keeping the comments as is